pandas/tests/indexes/test_setops.py

```diff
@@ -95,3 +103,38 @@ def test_union_dtypes(left, right, expected):
     b = pd.Index([], dtype=right)
     result = (a | b).dtype
     assert result == expected
+
+
+def test_union_duplicate_index_subsets_of_each_other():
```

Comment:
can you parameterize over index class here too

Reply:
Done